packages/typespec-ts/test-next/integration/serialization/serializeArray.test.ts

```diff
@@ -0,0 +1,318 @@
+    });
+
+    it("should handle a model array", async () => {
+      const { modelArray } = testArrays;
+      const result = serializeArray(modelArray);
+
+      const expected = await format(
+        `myArray.map((e: Foo)=>({...(e),
+                            "bar": ({...(e["bar"]),
+                            "baz": ({...(e["bar"]["myBaz"]),
+                            "qux": (e["bar"]["myBaz"]["qux"])})})}))`,
+        { parser: "typescript" }
+      );
+
+      const actual = await format(result, { parser: "typescript" });
+
+      expect(actual).to.equal(expected);
+    });
+
+    // Handle cyclic models
+    it.skip("should handle a cyclic array", async () => {
+      const { cyclicArray } = testArrays;
+      const result = serializeArray(cyclicArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle an optional properties array", async () => {
+      const { optionalPropertiesArray } = testArrays;
+      const result = serializeArray(optionalPropertiesArray);
+
+      const expected = await format(
+        `myArray.map((e: OptionalFoo) => ({
+                            ...e,
+                            ...(e["bar"] === undefined
+                                ? {}
+                                : {
+                                    bar: {
+                                    ...e["bar"],
+                                    baz: { ...e["bar"]["myBaz"], qux: e["bar"]["myBaz"]["qux"] },
+                                    },
+                                }),
+                            }));`,
+        { parser: "typescript" }
+      );
+
+      const actual = await format(result, { parser: "typescript" });
+
+      expect(actual).to.equal(expected);
+    });
+  });
+
+  describe("deserialization", () => {
+    let testArrays: TestArrays;
+
+    beforeEach(async () => {
+      testArrays = await getSdkArrays(runner, "deserialize");
+    });
+
+    it("should handle a string array", async () => {
+      const { stringArray } = testArrays;
+      const result = serializeArray(stringArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle a numeric array", async () => {
+      const { numericArray } = testArrays;
+      const result = serializeArray(numericArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle a boolean array", async () => {
+      const { booleanArray } = testArrays;
+      const result = serializeArray(booleanArray);
+
+      expect(result).to.equal("myArray");
+    });
+
+    it("should handle a date array", async () => {
+      const { dateArray } = testArrays;
+      const result = serializeArray(dateArray);
+
+      expect(result).to.equal("myArray.map(Date)");
+    });
+
+    it("should handle a model array", async () => {
+      const { modelArray } = testArrays;
+      const result = serializeArray(modelArray);
+
+      const expected = await format(
+        `myArray.map((e: Foo)=>({...(e),
```

Comment:
Serializer and deserializer will have (e: Foo) however one refers to the Modular Foo and the other to the RLC foo. @dgetu are we handling this already or this needs a fix

Reply:
I was not setting up the serializerMap that's why I was getting the same name for RLC and Modular references